yarn-project/p2p/src/config.ts

```diff
@@ -174,12 +174,12 @@ export const p2pConfigMappings: ConfigMappingsType<P2PConfig> = {
     ...numberConfigHelper(1_000),
   },
   tcpListenAddress: {
-    env: 'TCP_LISTEN_ADDR',
+    env: 'P2P_TCP_LISTEN_ADDR',
```

Comment:
Oh wow. This probably explains why env vars weren't working as expected in the k8s tests- I needed to set both tcp and udp ports to `40400` for discv5 to work.